File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashMap = {}
        for i in range(len(nums)):
            need = target - nums[i]
            if need in hashMap:
                return [hashMap[need], i]
            hashMap[nums[i]] = i
        return

File: Data Structures & Algorithms/target-sum/submission-3.java
class Solution {
    public int findTargetSumWays(int[] nums, int target) {
        int sum = 0;
        for(int num : nums) {
            sum += num;
        }
        if ((sum + target) % 2 != 0) {
            return 0;
        }
        int goal = (sum + target) / 2;
        if (goal < 0) return 0;
        int[] dp = new int[goal + 1];
        dp[0] = 1;
        for(int num: nums) {
            for(int j = goal; j >= num; j--) {
                dp[j] += dp[j - num];
            }
        }
        return dp[goal];
    }
}
